packages/riverpod_lint/bin/custom_lint.dart

```diff
@@ -15,8 +15,14 @@ import 'package:riverpod_lint/src/type_checker.dart';
 
 const _providerBase =
     TypeChecker.fromName('ProviderBase', packageName: 'riverpod');
-const _autoDispose =
-    TypeChecker.fromName('AutoDisposeProviderBase', packageName: 'riverpod');
+const _nonAutoDispose = TypeChecker.any([
```

Comment:
Could you rename it to `_alwaysAliveProviderListenable`?

That'd avoid the double negation of `(!_nonAutoDispose)`